---
title: "mitoClone2: Supplementary Tutorial"
author: "Benjamin Story, Ahrmad Annan"
date: "`r format(Sys.time(), '%d %B %Y')`"
output: 
  pdf_document: 
    keep_tex: yes
    toc: true
    number_sections: true
vignette: >
  %\VignetteEncoding{UTF-8}
  %\VignetteIndexEntry{"mitoClone2: Discovering bonafide mitochondrial variants in RNA-seq data."}
  %\VignetteEngine{knitr::knitr}
---

# References {-}
Corces, M., Buenrostro, J., Wu, B. _et al._ Lineage-specific and single-cell chromatin accessibility charts human hematopoiesis and leukemia evolution. _Nat Genet_ __48__, 1193â€“1203 (2016). https://doi.org/10.1038/ng.3646. [PMID: 27526324](https://pubmed.ncbi.nlm.nih.gov/27526324/)


# Demonstration of mitochondrial variant discovery from RNA-sequencing data
As a proof of concept, we demonstrate the ability of the mitoClone2 package to extract _bona fide_ private mutations from paired bulk ATAC-seq and RNA-seq samples. The goal here was to verify that variants in the mtDNA, which are specific to individual populations of cells, can be readily detected using RNA-seq data alone. We downloaded public bulk ATAC-seq and RNA-seq experimental data which were performed in parallel on cancer patients.

The data used for this analysis, from Corces _et al._ (cited in **References**), can be downloaded from the NCBI Gene Expression Omnibus (GEO) which is accessible at: https://www.ncbi.nlm.nih.gov/geo/query/acc.cgi?acc=GSE74912

The RNA-seq reads were aligned to the _Homo sapiens_ hg38 genome (Ensembl release 95) using `STAR v2.7.2b` with default parameters.
The ATAC-seq reads were trimmed and aligned to the _Homo sapiens_ hg38 genome using `bowtie2 v2.3.0` with the following parameters: `-X 2000`.

Count tables for each sample were tabulated using the `baseCountsFromBamList` a function included in the mitoClone2 package. The sample name for each putative patient was inferred from the GEO metadata. Only the following patients were included: `6792, 1022, 2596, 5483, 6926, 7256, SU048, SU070, SU209, SU351, SU353, SU444, SU484, SU496, SU501, SU575, SU583, and SU654` as these patients had both DNA (i.e. ATAC-seq) and RNA data available.

We note here that although this is bulk data, we can still treat each sample as if it were a single cell.

We note at this point that there are two possible filtering methods that can be used for downstream analysis.

### Cohort Method:
* Preferred for controlling wet-lab variability by using multiple independent single-cell datasets, generated using the same sequencing method, to exclude shared variants and identify patient/time-point specific variants.
* The excluded variants could be biological (common polymorphisms) or technical (artifacts from specific reagents or bioinformatics tools).
* This method helps filter out both technical and biological noise, by determining what signal can be attributed to background noise.

### Exclusion Lists Method:
* Utilizes established lists of known artifactual or challenging genomic regions, encompassing three sets:
  - Known RNA-editing sites in mitochondria from REDIportal (PMID: 27587585)
  - Regions flanking homopolymer stretches of 3 or more nucleotides
  - Shared variants specific to a given sequencing method
* The latter set of shared variants, included in the package, is derived from datasets generated using a modified Smart-seq2.
* This method is ideal for users who lack access to multiple datasets or require a fast, predefined filtering option.

The following sections provide a detailed walkthrough of both filtering methods, starting with the *Cohort* method.


# Complete workflow

## Load packages



```{r, eval=FALSE}
library(mitoClone2)
library(reshape2)
library(viridis)
library(ComplexHeatmap)
library(ggplot2)
```
```{r, eval=TRUE, include=FALSE}

library(mitoClone2)
library(ggplot2)
library(reshape2)
library(viridis)
library(ComplexHeatmap)
library(knitr)

opts_chunk$set(tidy.opts=list(width.cutoff=70),tidy=TRUE)
```


## Prepare some helper functions

A variety of extra functions are provided to minimize code redundancy.

```{r, eval=TRUE}
#-- Function that converts variant string format from "X1337O.G" to "1337 O>G"
fixvarName <- function(x){ gsub("\\.",">",gsub("^X","",x)) }

#-- Function that takes a dataset and name and gets mean allele freq. per patient
getptMean <- function(Data,name){
  df <- data.frame(t(Data))
  df.l <- lapply(split(df,gsub("\\-.*","",row.names(df))),colMeans)
  df.l <- melt(do.call(cbind,df.l))
  colnames(df.l) <- c('var','Patient',name)
  return(df.l)
}
```

This vignette uses data that is large (~30 MB) and thus not included with the package, but it can be downloaded separately via GitHub at: https://github.com/benstory/mitoClone2_supplemental.

The **mitoClone2** package provides some sample datasets. However, to emphasize the ability of mitoClone2 to detect real mitochondrial variants in RNA-seq data alone, we opted to use an additional more appropriate dataset where orthogonal variant confirmation via ATAC-seq was possible. These allele count tables, imported below, were generated by the `baseCountsFromBamList` function - for ATAC-seq the `bam2R` quality threshold was set to 30 instead of the default.

```{r, eval=TRUE}
#-- Here is an example of how we would generate our allele count files locally - SKIPPED
## vars.rna <- baseCountsFromBamList(bamfiles = list.files('corces_RNA','\\.bam$',full = TRUE), sites = "chrM:1-15000")
#the files need to be located in R's working directory!
#-- Download the external data from the mitoClone2 Supplemental GitHub Page
SuppGitURL <- "https://github.com/benstory/mitoClone2_supplemental/raw/main/"
##system(paste0("wget -nv ", SuppGitURL,'corces_nt_counts_per_position_rna.RDS'))
##system(paste0("wget -nv ", SuppGitURL,'corces_nt_counts_per_position_quality30_atac.RDS'))
#-- Loading the external data from GSE74246 (rna)
vars.rna <- readRDS('corces_nt_counts_per_position_rna.RDS')
#-- Loading the external data from GSE74912 (atac)
vars.atac <- readRDS('corces_nt_counts_per_position_quality30_atac.RDS')
```

We run mutationCallsFromCohort on the RNA-seq allele count tables for each of these patient samples using default parameters excepting for `MINFRAC.OTHER=0.2, MINREADS=500, MINCELL=1, MINCELLS.PATIENT=1`.These parameters are modified from the default values because we are not dealing with single-cell data. The `MINCELL` parameter now refers to the total number of samples from each individual patient. Given that there are two or more samples per patient, we require that the variants be present in at least a single sample.

The resulting 153 variants from this command were further filtered to remove 8 universal variants with an allele frequency over 5\% in more than 6 RNA-seq samples (i.e. non-patient exclusive mutations) leaving 145 putative variants of interest.

## Call variants from the RNA-seq data
```{r, eval=TRUE, message=FALSE}
vars.rna.call <- mutationCallsFromCohort(vars.rna,MINFRAC=0.1,MINFRAC.PATIENT=0.01,MINFRAC.OTHER=0.2,MINREADS=500,MINCELL=1,MINCELLS.PATIENT=1,patient=gsub("\\-.*","",names(vars.rna)),genome='hg38',sites='chrM:1-16569',USE.REFERENCE=TRUE)
```

## Examine the excluded universally shared variants
```{r, eval=TRUE}
#-- We extract the list of excluded sites and save for downstream analysis 
vars.exclusions <- vars.rna.call[['exclusionlist']]
head(vars.exclusions)
vars.rna.call[['exclusionlist']] <- NULL
```

## Extract putative private variants
```{r, eval=TRUE}
#-- Pull all the identified mutations that are likely exclusive to patients
candidates <- unlist(lapply(vars.rna.call,function(x) names(x@cluster) ))

#-- Remove variants that appear in clusters separated by less than 3bp for each patient
candidates <- removeWindow(fixvarName(sort(unique(candidates))),window=3)

#-- Extract counts of putative candidates from the RNA-seq
rna.counts <- pullcountsVars(vars.rna, candidates)
plotData <- rna.counts$M/(rna.counts$M + rna.counts$N)

#-- Remove variants that seem to be universal (i.e. non-patient exclusive)
#-- Our cutoff is an allele frequency over 5% in more than 6 samples
plotData <- plotData[rowSums(plotData > 0.05) <= 6,]
print(NROW(plotData))
```

## Prepare metadata for the downstream plotting

Extract patient identifiers and set colors.

```{r, eval=TRUE}
#-- Extract patient IDs
pts <- gsub("\\-.*","",colnames(plotData))
#-- Set unique colors for each patient
pts.colors <- c("#FF7F00","brown","palegreen2","khaki2","#FB9A99","darkorange4","green1","gold1","steelblue4","dodgerblue2","darkturquoise","orchid1","black","#6A3D9A","blue1","#CAB2D6","gray70","yellow3")
names(pts.colors) <- levels(factor(pts))
```

Prepare metadata for samples and heatmaps.

```{r, eval=TRUE}
#-- Preparing a metadata data.frame that includes:
#-- the sample name
#-- the patient associated to the sample
#-- the technology used to generate the sample
meta.data <- melt(c(names(vars.rna),names(vars.atac)))
row.names(meta.data) <- meta.data$value
meta.data$Patient <- gsub("\\-.*","",meta.data$value)
meta.data$Method <- paste0(gsub(".*\\_","",meta.data$value),"-seq")
meta.data$value <- NULL
```

## Investigate allele frequencies of putative private variants across sequencing technologies

We then extracted these same variants from both the ATAC-seq and RNA-seq (above) samples using the `pullcountsVars` function.

Prepare a heatmap of the RNA-seq data.

```{r, eval=TRUE}
#-- Setup heatmap annotation - plotData
ha_column = HeatmapAnnotation(df=meta.data[colnames(plotData),'Patient',drop=FALSE], col=list(Patient=pts.colors),show_annotation_name = FALSE)
#-- Build a heatmap of the RNA-seq allele frequencies for each variant
hrna <- Heatmap(plotData,col=viridis(100,option='plasma'),show_row_names=TRUE,show_column_names=FALSE,top_annotation=ha_column,column_title = "RNA-seq",show_column_dend = FALSE,show_row_dend = FALSE,column_title_gp = gpar(fontsize = 24),row_names_gp = gpar(fontsize = 2), name='Allele\nfreq.')
```

Prepare a heatmap of the ATAC-seq data.

```{r, eval=TRUE}
#-- Extract counts of putative candidates from the ATAC-seq
atac.counts <- pullcountsVars(vars.atac, row.names(plotData))
plotData.atac <- atac.counts$M / (atac.counts$M + atac.counts$N)

#-- Setup heatmap annotation for plotData.atac
ha_column = HeatmapAnnotation(df=meta.data[colnames(plotData.atac),'Patient',drop=FALSE], col=list(Patient=pts.colors))
#-- Build a heatmap of the ATAC-seq allele frequencies for each variant
hatac <- Heatmap(plotData.atac,col=viridis(100,option='plasma'),show_row_names=TRUE,show_column_names=FALSE,top_annotation=ha_column,column_title = "ATAC-seq",show_column_dend = FALSE,show_row_dend = FALSE,column_title_gp = gpar(fontsize = 24),row_names_gp = gpar(fontsize = 2), name='Allele\nfreq.')
```

The variants identified seem to allow clear differentiation of the individual patients across technologies.

```{r, eval=TRUE, warning=FALSE}
#-- Plot heatmaps for both technologies side-by-side - S3
hrna+hatac
```
\linebreak
__Supplementary Figure S3.__ Allele frequencies of mitochondrial variants identified as patient-specific from RNA-seq.

## Verifying consistency between ATAC-seq and RNA-seq for putative private variants

Extract the mean allele frequencies per patient for each variant.

```{r, eval=TRUE}
#-- Extract mean allele freq per patient per technology and merge
df.rna <- getptMean(plotData,'RNA-seq')
df.atac <- getptMean(plotData.atac,'ATAC-seq')

#-- Sanity check all data.frames should be similar
all(c(identical(df.rna$Patient,df.atac$Patient)),identical(df.rna$var,df.atac$var))

#-- merge into a new data.frame
df.both <- df.rna
df.both$'ATAC-seq' <- df.atac$'ATAC-seq'

#-- Perform a correlation test on the samples with Pearson
cor_both <- cor.test(df.both$ATAC, df.both$RNA)
colnames(df.rna)[2] <- 'Patient'
```

Plot the correlation between the RNA-seq and the ATAC-seq.

```{r, eval=TRUE}
#-- Plot the correlation and allele frequency comparison - S4
ggplot(df.both,aes(x=`RNA-seq`,y=`ATAC-seq`,color=Patient)) + geom_point(size=4) + theme_classic(base_size=14) + geom_abline(slope=1,color='grey',linetype='dashed') + scale_colour_manual(values=pts.colors) + annotate('text',x=0.25,y=0.80,label=paste0('r = ',cor_both$estimate)) + ylab('ATAC-seq\nAllele frequency') + xlab('RNA-seq\nAllele frequency') + theme(plot.margin=unit(c(1,1,1,1),"cm"),legend.text=element_text(size=8)) + guides(color = guide_legend(ncol = 2))
```
\linebreak
__Supplementary Figure S4.__ Allele frequencies of RNA-seq compared to ATAC-seq with Pearson correlation coefficient.

Merge your two large allele frequency data.frames.
```{r, eval=TRUE}
#-- Merge the RNA-seq and ATAC-seq allele frequency data.frames
merged.vars <- merge(plotData.atac,plotData, by="row.names", all = T)
row.names(merged.vars) <- merged.vars$Row.names
merged.vars$Row.names <- NULL
```

Calculate the Euclidean distance between samples and prepare metadata.
```{r, eval=TRUE}
#-- Create distance matrix from merged method data.frame
#-- The default method is set to 'euclidean'
distmat <- data.frame(as.matrix(dist(t(merged.vars))))
colnames(distmat) <- row.names(distmat)
distance.colors <- list(Method=c("RNA-seq"="red","ATAC-seq"="blue"),Patient=pts.colors)
ha_column <- HeatmapAnnotation(df = meta.data[colnames(distmat),],col=distance.colors)
```

Plot the distance-matrix heatmap.
```{r, eval=TRUE, warning=FALSE}
#-- S5
Heatmap(distmat,col=viridis(100),show_row_names=FALSE,show_column_names=FALSE,top_annotation=ha_column,show_column_dend = FALSE,show_row_dend = FALSE,name='Euclidean\ndistance')
```
\linebreak
__Supplementary Figure S5.__ Clustered distance matrix of combined variant allele frequencies from ATAC-seq and RNA-seq. 

## Identify variants resulting from cancer evolution

Even in this bulk sequencing experiment, we see evidence of variants that differ between the cancerous and pre-cancerous states.

Let's home in on an example, patient SU444, and extract their unique variants.

```{r, eval=TRUE}
su444 <- fixvarName(names(vars.rna.call[['SU444']]@cluster))
print(su444)
```

Pull the corresponding variant allele frequencies from both the ATAC-seq and RNA-seq datasets and prepare to plot.

```{r, eval=TRUE}
#-- Extract raw AFs from the datasets
su444rna <- data.frame(plotData[su444,grep('SU444',colnames(plotData))])
su444atac <- data.frame(plotData.atac[su444,grep('SU444',colnames(plotData.atac))])

#-- Merge into a single data.frame and print
su444 <- cbind(su444rna,su444atac)
print(su444)
su444_ann <- HeatmapAnnotation(df = subset(meta.data,Patient == 'SU444'), col = list(Method = c("RNA-seq" = "red", "ATAC-seq" = "blue"), Patient = pts.colors))
```

Plot the variant allele frequencies for this patient.

```{r, eval=TRUE}
#-- S6
Heatmap(as.matrix(su444), col=viridis(100,option='plasma'),show_row_names=TRUE,show_column_names=TRUE,top_annotation=su444_ann,column_split = rep(c("RNA-seq", "ATAC-seq"), each=3),show_column_dend = FALSE,show_row_dend = FALSE, name='Allele\nfreq.')
```
\linebreak
__Supplementary Figure S6.__ A zoomed-in version of (**Supplementary Figure S3**), focused on variants specific to patient SU444.

We can also attempt to infer a cancer phylogenetic tree (CPT) relating the different mutations and illustrate their locations using the built-in `mitoPlot` function.


```{r, eval=FALSE}
#-- If you want to reproduce the final figure (S7) you need to run this step
#-- However the figure here (below) has undergone significant post-processing
#-- Use SCITE to infer a CPT for patient SU444
clust444 <- varCluster(vars.rna.call[['SU444']],method='SCITE')

#-- S7 - bottom
mitoPlot(fixvarName(names(vars.rna.call[['SU444']]@cluster))[c(1,3,1,2)],patient=rep(rev(c('pHSC','Blast/LSC')),each=2),showLegend=TRUE)
```
![title](S7.pdf)\linebreak
__Supplementary Figure S7.__ The results of running `SCITE` (top) and the `mitoPlot` function (bottom) on the RNA-seq variant counts from patient SU444. The tree illustrates a potential phylogenetic relationship underlying cancerous clonal evolution differentiating the pre-leukemic (pHSC) sample from the blast/leukemic stem cell (LSC) samples. 


# Alternate workflow with Exclusionlist filtering method 

To complete our proof of concept, we apply a different universal variant filtering method based on existing exclusion lists that are shipped with the mitoClone2 package. This filtering method should only be used when the dataset includes only a single sample from one individual/time-point.

We re-import the allele count tables.

```{r, eval=TRUE}
#-- Here is an example of how we would generate our allele count files locally - SKIPPED
## vars.rna <- baseCountsFromBamList(bamfiles = list.files('corces_RNA','\\.bam$',full = TRUE), sites = "chrM:1-15000")
#the files need to be located in R's working directory!
#-- Loading the external data from GSE74246 (rna)
vars.rna <- readRDS('corces_nt_counts_per_position_rna.RDS')
#-- Loading the external data from GSE74912 (atac)
vars.atac <- readRDS('corces_nt_counts_per_position_quality30_atac.RDS')
```

We run mutationCallsFromExclusionlist on the RNA-seq allele count tables for each of these patient samples using default parameters to call mutations using the exclusion list included in the package.

## Call variants from the RNA-seq data using provided Exclusionlists
```{r, eval=TRUE}
# calling mutations using our exclusionlist
vars.rna.call <- mutationCallsFromExclusionlist(vars.rna)
```

## Downstream analysis similar to 2.5-2.9

From there, we will use the same downstream analysis as with the variant calling using the cohort filtering method.

```{r, eval=TRUE, results='hide', fig.show='hide', warning=FALSE}
#-- Pull all the identified mutations that are likely exclusive to patients
candidates <- sub(">", "",fixvarName(sort(unique(names(vars.rna.call@cluster)))), fixed = TRUE)

#-- Remove variants that appear in clusters separated by less than 3bp for each patient
candidates <- removeWindow(candidates,window=3)

#-- Extract counts of putative candidates from the RNA-seq
rna.counts <- pullcountsVars(vars.rna, candidates)
plotData <- rna.counts$M/(rna.counts$M + rna.counts$N)

#-- Remove variants that seem to be universal (i.e. non-patient exclusive)
#-- Our cutoff is an allele frequency over 5% in more than 6 samples
plotData <- plotData[rowSums(plotData > 0.05) <= 6,]

#-- Extract patient IDs
pts <- gsub("\\-.*","",colnames(plotData))
#-- Set unique colors for each patient
pts.colors <- c("#FF7F00","brown","palegreen2","khaki2","#FB9A99","darkorange4","green1","gold1","steelblue4","dodgerblue2","darkturquoise","orchid1","black","#6A3D9A","blue1","#CAB2D6","gray70","yellow3")
names(pts.colors) <- levels(factor(pts))

#-- Preparing a metadata data.frame that includes:
#-- the sample name
#-- the patient associated to the sample
#-- the technology used to generate the sample
meta.data <- melt(c(names(vars.rna),names(vars.atac)))
row.names(meta.data) <- meta.data$value
meta.data$Patient <- gsub("\\-.*","",meta.data$value)
meta.data$Method <- paste0(gsub(".*\\_","",meta.data$value),"-seq")
meta.data$value <- NULL

#-- Setup heatmap annotation - plotData
ha_column = HeatmapAnnotation(df=meta.data[colnames(plotData),'Patient',drop=FALSE], col=list(Patient=pts.colors),show_annotation_name = FALSE)
#-- Build a heatmap of the RNA-seq allele frequencies for each variant
hrna <- Heatmap(plotData,col=viridis(100,option='plasma'),show_row_names=TRUE,show_column_names=FALSE,top_annotation=ha_column,column_title = "RNA-seq",show_column_dend = FALSE,show_row_dend = FALSE,column_title_gp = gpar(fontsize = 24),row_names_gp = gpar(fontsize = 2), name='Allele\nfreq.')

#-- Extract counts of putative candidates from the ATAC-seq
atac.counts <- pullcountsVars(vars.atac, row.names(plotData))
plotData.atac <- atac.counts$M / (atac.counts$M + atac.counts$N)

#-- Setup heatmap annotation for plotData.atac
ha_column = HeatmapAnnotation(df=meta.data[colnames(plotData.atac),'Patient',drop=FALSE], col=list(Patient=pts.colors))
#-- Build a heatmap of the ATAC-seq allele frequencies for each variant
hatac <- Heatmap(plotData.atac,col=viridis(100,option='plasma'),show_row_names=TRUE,show_column_names=FALSE,top_annotation=ha_column,column_title = "ATAC-seq",show_column_dend = FALSE,show_row_dend = FALSE,column_title_gp = gpar(fontsize = 24),row_names_gp = gpar(fontsize = 2), name='Allele\nfreq.')

#-- Plot heatmaps for both technologies side-by-side
hrna+hatac

#-- Extract mean allele freq per patient per technology and merge
df.rna <- getptMean(plotData,'RNA-seq')
df.atac <- getptMean(plotData.atac,'ATAC-seq')

#-- Sanity check all data.frames should be similar
all(c(identical(df.rna$Patient,df.atac$Patient)),identical(df.rna$var,df.atac$var))

#-- merge into a new data.frame
df.both <- df.rna
df.both$'ATAC-seq' <- df.atac$'ATAC-seq'

#-- Perform a correlation test on the samples with Pearson
cor_both <- cor.test(df.both$ATAC, df.both$RNA)
colnames(df.rna)[2] <- 'Patient'
```

```{r, eval=TRUE, results='hide', fig.show='hide', warning=FALSE}
#-- Plot the correlation and allele frequency comparison 
ggplot(df.both,aes(x=`RNA-seq`,y=`ATAC-seq`,color=Patient)) + geom_point(size=4) + theme_classic(base_size=14) + geom_abline(slope=1,color='grey',linetype='dashed') + scale_colour_manual(values=pts.colors) + annotate('text',x=0.25,y=0.80,label=paste0('r = ',cor_both$estimate)) + ylab('ATAC-seq\nAllele frequency') + xlab('RNA-seq\nAllele frequency') + theme(plot.margin=unit(c(1,1,1,1),"cm"),legend.text=element_text(size=8)) + guides(color = guide_legend(ncol = 2))
```


```{r, eval=TRUE, results='hide', warning=FALSE}
#-- Merge the RNA-seq and ATAC-seq allele frequency data.frames
merged.vars <- merge(plotData.atac,plotData, by="row.names", all = T)
row.names(merged.vars) <- merged.vars$Row.names
merged.vars$Row.names <- NULL

#-- Create distance matrix from merged method data.frame
#-- The default method is set to 'euclidean'
distmat <- data.frame(as.matrix(dist(t(merged.vars))))
colnames(distmat) <- row.names(distmat)
distance.colors <- list(Method=c("RNA-seq"="red","ATAC-seq"="blue"),Patient=pts.colors)
ha_column <- HeatmapAnnotation(df = meta.data[colnames(distmat),],col=distance.colors)

#-- S8
Heatmap(distmat,col=viridis(100),show_row_names=FALSE,show_column_names=FALSE,top_annotation=ha_column,show_column_dend = FALSE,show_row_dend = FALSE,name='Euclidean\ndistance')
```

__Supplementary Figure S8.__ Clustered distance matrix of combined variant allele frequencies from ATAC-seq and RNA-seq, using Exclusionlist filtering method.

## The *Exclusionlist* filtering method compared to the *Cohort* method

The Exclusionlist variant filtering approach increased false negative hits and slightly decreased the correlation between RNA-seq and ATAC-seq allele frequencies. It does not work as well as the cohort method for filtering shared variants but still can be used with confidence to identify clones when no other option is available.

In conclusion, we suggest that users always use the cohort method whenever possible (i.e. access to multiple similar datasets).

# Adding metadata to Seurat objects

Below is just an example where we add the variant allele frequencies calculated above to the minimal test dataset included with Seurat for illustrative purposes.

```{r, eval=TRUE, results='hide', warning=FALSE, out.height = "256px", message=FALSE}
library(Seurat)    

## randomly assign our labels to cells
set.seed(123)
colnames(pbmc_small)[seq(colnames(plotData))] <- sample(colnames(plotData))
 
## add the VAF information
pbmc_small <- AddMetaData(pbmc_small,col.name = 'Variant AF',metadata = plotData['1336G>A',])

## plot the default tSNE and the VAF colored one
DimPlot(pbmc_small) + FeaturePlot(pbmc_small,features = 'Variant AF', cols=viridis(2))
```

```{r label='Session information', eval=TRUE, echo=FALSE}
sessionInfo()
```

